.github/workflows/pull-request.yml

```
@@ -45,216 +49,409 @@ jobs:
     steps:
-      - name: Checkout code
+      - name: Checkout
         uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
 
-      - name: Get Changed Unauthorized files
-        id: changed-unauth-files
-        uses: tj-actions/changed-files@v40
+      - name: Set up Python 3.11
+        uses: actions/setup-python@v4
```

actionlint (1.7.8)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)